vernac/comAssumption.ml

```diff
@@ -180,3 +180,31 @@ let do_assumptions kind nl l =
       in
       subst'@subst, status' && status, next_uctx uctx)
     ([], true, uctx) l)
+
+let declare_register id (t,ctx) o_t imps =
+  let kn =
+    declare_constant id (PrimitiveEntry((t,ctx),o_t),IsPrimitive) in
+  maybe_declare_manual_implicits false (ConstRef kn) imps;
+  register_message id
+
+let do_register id r =
+  match r with
+  | RegisterInline ->
+    begin match Constrintern.global_reference id.CAst.v with
+    | ConstRef c -> Global.register_inline c
+    | _ -> CErrors.user_err (Pp.str "Register Inline: expecting a constant")
+    end
+  | RegisterPrimitive(t,op) ->
+     if Dumpglob.dump () then Dumpglob.dump_definition id false "ax";
+     let env = Global.env () in
+     let ienv = empty_internalization_env in
+     let evd, (t,imps) = interp_assumption (Evd.from_env env) env ienv [] t in
+     let uvars = Univops.universes_of_constr env t in
```

Comment:
You should probably do a universe minimization before universes_of_constr.

Reply:
Oh! I thought I had adapted existing code. Will check.

Reply:
Also after 886a9c2fb2 interp_assumption returns an econstr so you will have to to_constr after the minimization.

Reply:
> You should probably do a universe minimization before universes_of_constr.

@SkySkimmer what function should I call exactly?

Reply:
Evd.nf_constraints

Reply:
Fixed, thanks a lot!